### PR TITLE
Feature/IO-653_share_plugin

### PR DESCRIPTION
**Pull request by egor-slatvinskiy-idf**

### CODE REVIEW

File: packages/presentation/lib/l10n/intl_ru.arb [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+  "share": "Смотри, какой я фильм нашел! \n\nhttps://www.themoviedb.org/movie/{movie_id}?language={language}"
+}
```
Comment (vladimir-hodaryonok-idf) [inline packages/presentation/lib/l10n/intl_ru.arb:3]: Did u tested ru locale? My app crushed when I switched on to ru, cuz require all strings to be ru.

File: share_android_ios/lib/share_android_ios.dart [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import 'package:flutter/services.dart';
+
+class ShareAndroidIos {
+  static const MethodChannel _channel =
+      const MethodChannel('share_android_ios');
+
+  static Future<void> share(String message) async {
+    await _channel.invokeMethod('share', {'message': message});
```
Comment (vladimir-hodaryonok-idf) [inline share_android_ios/lib/share_android_ios.dart]: The call might fail—for example, if the platform doesn’t support the platform API (such as when running in a simulator), so wrap the invokeMethod call in a try-catch statement. (C) flutter docs

File: share_android_ios/lib/share_android_ios_method_channel.dart [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+class MethodChannelShareAndroidIos extends ShareAndroidIosPlatform {
+  @visibleForTesting
+  final methodChannel = const MethodChannel('share_android_ios');
+
+  @override
+  Future<String?> getPlatformVersion() async {
+    final version = await methodChannel.invokeMethod<String>('getPlatformVersion');
+    return version;
+  }
+}
```
Comment (vladimir-hodaryonok-idf) [inline share_android_ios/lib/share_android_ios_method_channel.dart]: do you need this? Haven't seen any implementation in [ShareAndroidIosPlugin.java] for "getPlatformVersion"

File: share_android_ios/lib/share_android_ios_platform_interface.dart [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+  
+  static set instance(ShareAndroidIosPlatform instance) {
+    PlatformInterface.verifyToken(instance, _token);
+    _instance = instance;
+  }
+
+  Future<String?> getPlatformVersion() {
+    throw UnimplementedError('platformVersion() has not been implemented.');
+  }
+}
```
Comment (vladimir-hodaryonok-idf) [inline share_android_ios/lib/share_android_ios_platform_interface.dart]: -//-

File: packages/data/pubspec.yaml [hunk 5]
```diff
@@ -17,7 +17,6 @@ dependencies:
   firebase_auth: ^3.4.0
   cloud_firestore: ^3.2.0
   flutter_facebook_auth: ^4.4.1+1
-  flutter_facebook_login: ^3.0.0
```
Comment (vladimir-hodaryonok-idf) [inline packages/data/pubspec.yaml:20]: why do you delete it?

**Review by vladimir-hodaryonok-idf — APPROVED** [review]

File: packages/presentation/lib/l10n/intl_ru.arb [hunk 6]
```diff
@@ -0,0 +1,3 @@
+{
+  "share": "Смотри, какой я фильм нашел! \n\nhttps://www.themoviedb.org/movie/{movie_id}?language={language}"
```
Comment (elizaveta-seleznyova-idf) [inline packages/presentation/lib/l10n/intl_ru.arb:2]: why u have only one line here? In en version u have 16 variables

**Review by elizaveta-seleznyova-idf — APPROVED** [review]

File: packages/data/pubspec.yaml (hunk 5 above)
Comment (egor-slatvinskiy-idf) [reply packages/data/pubspec.yaml:20]: because we don't use it)

File: packages/presentation/lib/l10n/intl_ru.arb (hunk 1 above)
Comment (egor-slatvinskiy-idf) [reply packages/presentation/lib/l10n/intl_ru.arb:3]: yes, everything is adequate for me, only the share changes

File: packages/presentation/lib/l10n/intl_ru.arb (hunk 6 above)
Comment (egor-slatvinskiy-idf) [reply packages/presentation/lib/l10n/intl_ru.arb:2]: according to the assignment, we don’t have such a thing as adding a Russian locale to the entire application

File: share_android_ios/analysis_options.yaml [hunk 7]
```diff
@@ -0,0 +1 @@
+include: package:flutter_lints/flutter.yaml
```
Comment (alexei-kruk-idf) [inline share_android_ios/analysis_options.yaml]: Напишу на русском. Ты мог не создавать прям отдельный плагин в папке, а мог просто создать классы для работы с шерингом в папках ios и android и там-же их зарегать, но если тебе так больше нравится, то окей)

**Review by alexei-kruk-idf — APPROVED** [review]

**Review by yuliya-klescheva-idf — APPROVED** [review]

File: share_android_ios/analysis_options.yaml (hunk 7 above)
Comment (egor-slatvinskiy-idf) [reply share_android_ios/analysis_options.yaml]: В таске написано создать плагин, так я и создал:)

**Comment by sergei-mikhailovskii-idf** [comment]
let's move the plugin to the packages

File: packages/presentation/lib/l10n/intl_ru.arb (hunk 6 above)
Comment (elizaveta-seleznyova-idf) [reply packages/presentation/lib/l10n/intl_ru.arb:2]: according to this assignment we don't, but we have previous task about localization
It's look strange when u have a lot for English and only one line for Russian
How localization works when u put Russia location for your device?

File: packages/presentation/lib/ui/movie_details/bloc/movie_details_bloc.dart [hunk 8]
```diff
@@ -40,6 +45,14 @@ class _MovieDetailsBlocImpl
     _updateData();
   }
 
+  @override
+  void share(BuildContext context) {
```
Comment (sergei-mikhailovskii-idf) [inline packages/presentation/lib/ui/movie_details/bloc/movie_details_bloc.dart:49]: https://stackoverflow.com/questions/58505388/is-it-ok-to-pass-buildcontext-to-bloc

File: packages/share_android_ios/android/src/main/java/com/example/share_android_ios/ShareAndroidIosPlugin.java [hunk 9]
```diff
@@ -0,0 +1,73 @@
+package com.example.share_android_ios;
```
Comment (sergei-mikhailovskii-idf) [inline packages/share_android_ios/android/src/main/java/com/example/share_android_ios/ShareAndroidIosPlugin.java:1]: 1) apply formatting plz (cmd + option + L)
2) let's rewrite it in Kotlin

https://developer.android.com/kotlin/add-kotlin#convert

File: packages/share_android_ios/ios/Classes/ShareAndroidIosPlugin.m [hunk 10]
```diff
@@ -0,0 +1,72 @@
+#import "ShareAndroidIosPlugin.h"
```
Comment (sergei-mikhailovskii-idf) [inline packages/share_android_ios/ios/Classes/ShareAndroidIosPlugin.m:1]: and this in Swift

**Review by sergei-mikhailovskii-idf — CHANGES_REQUESTED** [review]

**Pull request merged**